src/helper.rs

```diff
@@ -222,7 +222,41 @@ where
     I: IntoIterator<Item = &'a A> + 'a,
 {
     let nftables = serde_json::to_string(nftables).expect("failed to serialize Nftables struct");
-    apply_ruleset_raw(&nftables, program, args)
+    apply_ruleset_raw(&nftables, program, args)?;
+    Ok(())
+}
+
+/// Apply the given rule set to the kernel, and returns the processed rule set with
+/// extra information.
+///
+/// This is done by using `nft --echo`. One can get rule handles from the returned
+/// objects for future modifications, positional inserts, as well as removal.
+pub fn apply_and_return_ruleset(nftables: &Nftables) -> Result<Nftables<'static>, NftablesError> {
+    apply_and_return_ruleset_with_args(nftables, DEFAULT_NFT, DEFAULT_ARGS)
+}
+
+/// Apply the given rule set by calling a custom `nft` with custom arguments, and
+/// returns the processed rule set with extra information.
+///
+/// This is done by using `nft --echo`. One can get rule handles from the returned
+/// objects for future modifications, positional inserts, as well as removal.
+pub fn apply_and_return_ruleset_with_args<'a, P, A, I>(
+    nftables: &Nftables,
+    program: Option<&P>,
+    args: I,
+) -> Result<Nftables<'static>, NftablesError>
+where
+    P: AsRef<OsStr> + ?Sized,
+    A: AsRef<OsStr> + ?Sized + 'a,
+    I: IntoIterator<Item = &'a A> + 'a,
+{
+    let nftables = serde_json::to_string(nftables).expect("failed to serialize Nftables struct");
+    let args = args
+        .into_iter()
+        .map(AsRef::as_ref)
+        .chain(Some("-e".as_ref()));
```

Comment:
```suggestion
        .chain(Some("--echo".as_ref()));
```
This matches the rustdoc explanation and more concise

Reply:
Done and rebased.